### PR TITLE
Add a note that the iterator must be fused for usage with `dropping`

### PR DESCRIPTION
**Pull request by tbu-**

### CODE REVIEW

**Comment by bluss** [comment]
must is too strong

**Comment by tbu-** [comment]
It will be incorrect if the iterator isn't fused and we hit the end. Should I phrase it like this?

**Comment by bluss** [comment]
That is much better.

**Comment by tbu-** [comment]
Changed.

File: src/lib.rs [hunk 1]
```diff
@@ -985,6 +985,9 @@ pub trait Itertools : Iterator {
     /// It works similarly to *.skip(* `n` *)* except it is eager and
     /// preserves the iterator type.
     ///
+    /// This will produce incorrect results if the iterator is not *fused* and
+    /// the number of elements in the iterator is less than `n`.
+    ///
```
Comment (bluss) [inline src/lib.rs]: Ok, sorry, can we add another correction? Dropping will work fine if n is larger than the number of elements in the iterator. The error would be to try to get more items from the returned iterator if n is larger than the number of elements.

**Comment by tbu-** [comment]
Changed. I'm not sure it's useful to provide this very long note instead of saying "This function assumes that the iterator is fused." and letting the user dive into the code if they want the exact behavior.

**Comment by bluss** [comment]
I can write a shorter note if you want.

**Comment by tbu-** [comment]
I'm not disappointed if you write your own note. :)

**Pull request closed without merging**